packages/dts-generator/lib/phases/json-to-ast.js

```diff
@@ -101,12 +101,27 @@ function buildVariable(property) {
  * @returns {FunctionDesc}
  */
 function buildFunction(ui5Method) {
-  assertKnownProps(["name", "parameters", "returnValue", "throws"], ui5Method);
+  assertKnownProps(
+    ["name", "parameters", "returnValue", "throws", "optional"],
+    ui5Method
+  );
+  // Simple check to avoid having the same parameter defined twice
+  const existingParameters = {};
+  const uniqueParameters = [];
+  if (ui5Method.parameters) {
+    ui5Method.parameters.forEach(parameter => {
+      // We need to prefix the parameter with _ otherwise parameter named constructor gets ignored :)
```

Comment:
I am not sure if prefixes are required.
Perhaps you want to use hasOwnProperty to distinguish the constructor from the prototype
and the constructor property (directly) on an object?